bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/impl/DynamicDeckServiceImpl.java

```diff
@@ -0,0 +1,224 @@
+     * <p>
+     * TODO Planning to create service account and remove the dependency of resourceResolver from this service
+     *
+     * @param collectionPath
+     * @param resourceResolver
+     * @return
+     */
+    @Override
+    public List<Resource> fetchAssetListFromCollection(String collectionPath, ResourceResolver resourceResolver) throws DynamicDeckDynamoException {
+        List<Resource> arrayList = new ArrayList<>();
+        if (StringUtils.isBlank(collectionPath)) {
+            throw new DynamicDeckDynamoException("Resource Path is Null/Empty ");
+        }
+        Resource collectionResource = resourceResolver.getResource(collectionPath);
+        if (collectionResource == null) {
+            throw new DynamicDeckDynamoException("No Resource exists at Supplied Path :" + collectionPath);
+        }
+
+        if (DamUtil.isSmartCollection(collectionResource)) {
+            return DynamicDeckUtils.fetchSmartCollectionResourceList(collectionResource);
+        }
+
+        Iterator<Asset> assets = DamUtil.getAssets(collectionResource);
+
+        assets.forEachRemaining(eachAsset -> {
+            Resource eachAssetResource = eachAsset.adaptTo(Resource.class);
+            if (null != eachAssetResource) {
+                arrayList.add(eachAsset.adaptTo(Resource.class));
+            }
+        });
+        return arrayList;
+    }
+
+    @Override
+    public List<Resource> fetchAssetListFromQuery(String queryString, ResourceResolver resourceResolver) {
+        Iterator<Resource> assetResourceIterator = resourceResolver.findResources(queryString, Query.JCR_SQL2);
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(assetResourceIterator, Spliterator.ORDERED),
+                false).collect(Collectors.toCollection(ArrayList::new));
+    }
+
+    @Override
+    public List<Resource> fetchAssetListFromTags(String tagsString, ResourceResolver resourceResolver) {
+        TagManager tagManager = resourceResolver.adaptTo(TagManager.class);
+        List<Resource> arrayList = new ArrayList<>();
+        String[] tagsArray = StringUtils.split(tagsString, DynamicDeckDynamoConstants.COMMA);
+        Arrays.stream(tagsArray).forEach(tag -> {
+            RangeIterator<Resource> tags = tagManager.find(tag);
+            tags.forEachRemaining(taggedResource -> {
+                if (taggedResource.getPath().startsWith(DynamicDeckDynamoConstants.DAM_ROOT) && taggedResource.getPath().endsWith(DynamicDeckDynamoConstants.SLASH + DynamicDeckDynamoConstants.DAM_METADATA)) {
+                    Asset taggedAsset = taggedResource.getParent().getParent().adaptTo(Asset.class);
+                    if (null != taggedAsset) {
+                        arrayList.add(taggedAsset.adaptTo(Resource.class));
+                    }
+                }
+            });
+        });
+        return arrayList;
+    }
+
+    private void addExportJobProperty(Asset assetObject, Object exportJobId) {
```

Comment:
as this is a private method, isn't it possible to make sure that the calling method does not provide null parameters?

Reply:
This second argument should be a `String`, no? AFAICT the only usage is from `Job.getId()` and that should be a `String`.